smartsim/settings/launchSettings.py

```diff
@@ -0,0 +1,407 @@
+        except KeyError:
+            raise ValueError(f"Invalid launcher type: {launcher}")
+        self._arg_translator = self._get_launcher()
+        
+        if env_vars:
+            validate_env_vars(env_vars)
+        self.env_vars = env_vars or {}
+
+        if launcher_args:
+            validate_args(launcher_args)
+        self.launcher_args = launcher_args or {}
+    
+    @property
+    def launcher(self):
+        return self._launcher
+
+    @property
+    def arg_translator(self):
+        return self._arg_translator
+
+    @property
+    def launcher_args(self) -> t.Dict[str, t.Union[int, str, float, None]]:
+        """Return an immutable list of attached launcher arguments.
+
+        :returns: attached run arguments
+        """
+        return self._launcher_args
+
+    @launcher_args.setter
+    def launcher_args(self, value: t.Dict[str, t.Union[int, str, float,None]]) -> None:
+        """Set the launcher arguments.
+
+        :param value: run arguments
+        """
+        self._launcher_args = copy.deepcopy(value)
+
+    @property
+    def env_vars(self) -> StringArgument:
+        """Return an immutable list of attached environment variables.
+
+        :returns: attached environment variables
+        """
+        return self._env_vars
+
+    @env_vars.setter
+    def env_vars(self, value: StringArgument) -> None:
+        """Set the environment variables.
+
+        :param value: environment variables
+        """
+        self._env_vars = copy.deepcopy(value)
+    
+    @property
+    def reserved_launch_args(self):
+        return self.arg_translator.set_reserved_launch_args()
+    
+    def _get_launcher(self) -> LaunchArgTranslator:
+        """ Map the Launcher to the LaunchArgTranslator
+        """
+        if self._launcher.value == 'slurm':
+            return SlurmArgTranslator()
+        elif self._launcher.value == 'mpiexec':
+            return MpiexecArgTranslator()
+        elif self._launcher.value == 'mpirun':
+            return MpiArgTranslator()
+        elif self._launcher.value == 'orterun':
+            return OrteArgTranslator()
+        elif self._launcher.value == 'alps':
+            return AprunArgTranslator()
+        elif self._launcher.value == 'lsf':
+            return JsrunArgTranslator()
+        elif self._launcher.value == 'pals':
+            return PalsMpiexecArgTranslator()
+        elif self._launcher.value == 'dragon':
+            return DragonArgTranslator()
+        elif self._launcher.value == 'local':
+            return LocalArgTranslator()
+
+    def launcher_str(self) -> str:
+        """ Get the string representation of the launcher
+        """
+        return self.arg_translator.launcher_str()
+    
+    def set_nodes(self, nodes: int) -> None:
+        """ Sets the number of nodes
+
+        :param nodes: The number of nodes
+        """
+
+        args = self.arg_translator.set_nodes(nodes)
+
+        if args:
+            self.update_launcher_args(args)
```

Comment:
per offline discussion: it was decided to have `self.arg_translator` manage its own state and to remove these pass through fns for better composability